lib/ansible/modules/system/iocage.py

```diff
@@ -0,0 +1,615 @@
+                               basejail=items[9]))
+        return output
+
+    @staticmethod
+    def _add_line_to_properties(properties: Dict, line: str):
+        """
+        Add a line of the output of 'iocage get' as an item to the properties dict
+        :param properties:
+        :param line:
+        """
+        items = line.split(":")
+        properties[items[0]] = items[1]
+
+
+def run_module(module: AnsibleModule, result: Dict):
+    iocage = IOCage(module, result, module.params.get('zpool'))
+    if iocage.zpool and not iocage.is_activated(iocage.zpool):
+        iocage.activate(iocage.zpool)
+        result['changed'] = True
+        message = str("ZPool %s activated" % iocage.zpool)
+        result['message'] = str("%s, %s" % (result['message'], message)) \
+            if result['message'] else message
+
+    try:
+        jail = Jail(name=module.params['name'],
+                    uuid=module.params['uuid'],
+                    release=module.params.get('release'),
+                    template=module.params.get('template'),
+                    empty=module.params.get('empty'),
+                    state=module.params.get('state'),
+                    properties=module.params.get('properties'))
+    except ValueError as ve:
+        module.fail_json(msg=str(ve))
+
+    if (jail.state == "present" or jail.state == "started") and not iocage.exists(jail):
+        iocage.create(jail)
+        result['changed'] = True
+        message = "Jail '%s' created" % jail.name
+        result['message'] = str("%s, %s" % (result['message'], message)) \
+            if result['message'] else message
+
+    if jail.state == "started" and not iocage.is_started(jail):
+        if iocage.has_changed_properties(jail):
+            iocage.set_properties(jail)
+        iocage.start(jail)
+        result['changed'] = True
+        message = str("Jail '%s' started" % jail.name)
+        result['message'] = str("%s, %s" % (result['message'], message)) \
+            if result['message'] else message
+
+    if jail.state == "started" and iocage.is_started(jail) and iocage.has_changed_properties(jail):
+        iocage.stop(jail)
+        iocage.set_properties(jail)
+        iocage.start(jail)
+        result['changed'] = True
+        message = str("Jail '%s' started" % jail.name)
+        result['message'] = str("%s, %s" % (result['message'], message)) \
+            if result['message'] else message
+
+    if jail.state == "present" and iocage.is_started(jail):
+        iocage.stop(jail)
+        if iocage.has_changed_properties(jail):
+            iocage.set_properties(jail)
+        result['changed'] = True
+        message = str("Jail '%s' stopped" % jail.name)
+        result['message'] = str("%s, %s" % (result['message'], message)) \
+            if result['message'] else message
+
+    if jail.state == "absent" and iocage.exists(jail):
+        if iocage.is_started(jail):
+            iocage.stop(jail)
```

Comment:
It would be nice to have a `force` option (`iocage stop --force`). Sometimes you might want to kill the container instead of shutting it down gracefully.

Reply:
Good idea, will implement it.

Reply:
Fixed by e75031f781d240dac23574e221f45dbf2ba9dc2d